src/Elastic.Apm/Report/Serialization/TrimmedStringJsonConverter.cs

```diff
@@ -1,12 +1,19 @@
 using System;
+using System.Runtime.CompilerServices;
 using Newtonsoft.Json;
 
 namespace Elastic.Apm.Report.Serialization
 {
 	internal class TrimmedStringJsonConverter : JsonConverter<string>
 	{
+		private readonly int _maxLength;
+
+		public TrimmedStringJsonConverter() : this(Consts.PropertyMaxLength) { }
+
+		public TrimmedStringJsonConverter(int maxLength) => _maxLength = maxLength;
```

Comment:
Similarly, a `// ReSharper disable once MemberCanBePrivate.Global` would turn off the suggestion which would break the code.

Reply:
fixed